### DISCUSSION TITLE
Put together a docker image for those who are interested

### DISCUSSION THREAD

**Discussion by gjreasoner** [livebud/bud#49] [Show and tell]
I put together a docker image to help build bud apps CI/CD style, might save people some time if they want some inspiration;

https://github.com/gjrdiesel/bud-docker

- I had a slight issue with the deployed binary saying it was missing `go.mod`, [I suggested packaging that up](https://github.com/gjrdiesel/bud-docker/blob/main/readme.md?plain=1#L16), will try to look into whats going on with that later (open to PRs!)
- It's probably much larger then it needs to be by using buster images but it gets the job done

[Once the port issue is decided](https://github.com/livebud/bud/discussions/42), I'll put a demo up of a running bud server on heroku, blocked at the moment I think by heroku expecting `PORT` env to be a single port.

Also if you plan on using a volume mount for this docker image, you might be [blocked until this issue is resolved](https://github.com/livebud/bud/issues/30).

Loving the framework so far Matt! Looking forward to learning more and thanks for sharing!

**Comment by matthewmueller** [comment]
Wow, thanks Justin! 🥰

> I had a slight issue with the deployed binary saying it was missing go.mod, [I suggested packaging that up](https://github.com/gjrdiesel/bud-docker/blob/main/readme.md?plain=1#L16), will try to look into whats going on with that later (open to PRs!)

Oh interesting, so `bud build` definitely relies on `go.mod` being present, but I wouldn't expect the `bud/app` binary to rely on it if that `bud/app` was built with `bud build` and not `bud run`. Not sure if that makes sense. Happy to clarify.

> It's probably much larger then it needs to be by using buster images but it gets the job done

Thanks for kicking this off. Ignition takes more energy than cruising. _(Just made that up, not even sure if that's true haha.)_

---

We'll get the linked issues resolved soon!

**Comment by syke99** [comment]
Nice work!!

You mentioned the image is pretty large at the moment due to using a buster image to build on top of. Just curious what caused the need for building in a buster image (I’m really new to Docker). Is it because of bud depending on Node.js? If so, it seems there projected path is to remove Node.js in [v0.3.0](https://github.com/livebud/bud/discussions/21#discussion-4071017). And then, maybe you could just use a go alpine image?
